Geometry/HcalAlgo/plugins/DDHCalXtalAlgo.cc

```diff
@@ -33,60 +38,72 @@ void DDHCalXtalAlgo::initialize(const DDNumericArguments & nArgs,
   iaxis      = int (nArgs["Axis"]);
   names      = vsArgs["Names"];
 
-  LogDebug("HCalGeom") << "DDHCalXtalAlgo debug: Parameters for positioning:: "
-		       << "Axis " << iaxis << "\tRadius " << radius 
-		       << "\tOffset " << offset << "\tDx " << dx << "\tDz " 
-		       << dz << "\tAngWidth " << angwidth/CLHEP::deg 
-		       << "\tNumbers " << names.size();
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalXtalAlgo::Parameters for positioning:"
+			       << " Axis " << iaxis << "\tRadius " << radius 
+			       << "\tOffset " << offset << "\tDx " << dx
+			       << "\tDz " << dz << "\tAngWidth " 
+			       << convertRadToDeg(angwidth) << "\tNumbers "
+			       << names.size();
   for (unsigned int i = 0; i < names.size(); i++)
-    LogDebug("HCalGeom") << "\tnames[" << i << "] = " << names[i];
-
+    edm::LogVerbatim("HCalGeom") << "\tnames[" << i << "] = " << names[i];
+#endif
   idNameSpace = DDCurrentNamespace::ns();
   idName      = sArgs["ChildName"]; 
-  DDName parentName = parent().name(); 
-  LogDebug("HCalGeom") << "DDHCalXtalAlgo debug: Parent " << parentName 
-		       << "\tChild " << idName << " NameSpace " << idNameSpace;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalXtalAlgo: Parent " << parent().name()
+			       << "\tChild " << idName << " NameSpace "
+			       << idNameSpace;
+#endif
 }
 
 void DDHCalXtalAlgo::execute(DDCompactView& cpv) {
 
   double theta[3], phi[3], pos[3];
   phi[0] = 0;
-  phi[1] = 90*CLHEP::deg;
-  theta[1-iaxis] = 90*CLHEP::deg;
+  phi[1] = 90._deg;
+  theta[1-iaxis] = 90._deg;
   pos[1-iaxis]   = 0;
   int number = (int)(names.size());
   for (int i=0; i<number; i++) {
     double angle = 0.5*angwidth*(2*i+1-number);
-    theta[iaxis] = 90*CLHEP::deg + angle;
+    theta[iaxis] = 90._deg + angle;
     if (angle>0) {
       theta[2]   = angle;
-      phi[2]     = 90*iaxis*CLHEP::deg;
+      phi[2]     = iaxis*90._deg;
     } else {
       theta[2]   =-angle;
-      phi[2]     = 90*(2-3*iaxis)*CLHEP::deg;
+      phi[2]     = (2-3*iaxis)*90._deg;
     }
     pos[iaxis]   = angle*(dz+radius);
-    pos[2]       = dx*abs(sin(angle)) + offset;
+    pos[2]       = dx*std::abs(sin(angle)) + offset;
   
     DDRotation rotation;
-    string rotstr = names[i];
+    std::string rotstr = names[i];
     DDTranslation tran(pos[0], pos[1], pos[2]);
     DDName parentName = parent().name(); 
 
-    if (abs(angle) > 0.01*CLHEP::deg) {
-      LogDebug("HCalGeom") << "DDHCalXtalAlgo test: Creating a new rotation " 
-			   << rotstr << "\t" << theta[0]/CLHEP::deg << "," 
-			   << phi[0]/CLHEP::deg << "," << theta[1]/CLHEP::deg 
-			   << "," << phi[1]/CLHEP::deg << "," 
-			   << theta[2]/CLHEP::deg << ","  << phi[2]/CLHEP::deg;
+    static const double tol = 0.000174523292; // 0.01 degree
```

Comment:
No need to change this line now, but wouldn't using `_deg` make this line look nicer?

`static constexpr double tol = 0.01_deg;`

Reply:
I shall remember and do that later